<?xml version="1.0"?>
<launch>

    <arg name="use_rviz" default="true" />

    <!--- Sim Time -->
    <param name="/use_sim_time" value="true" />

    <!--- Run Rviz -->
    <group if="$(arg use_rviz)">
        <node pkg="rviz" type="rviz" name="rviz" args="-d $(find points_preprocessor_usi)/launch/kitti.rviz" />
    </group>



    <!--- TF -->
    <node pkg="tf" type="static_transform_publisher" name="velodyne_to_velo_link"  args="0 0 0 0 0 0 /velo_link /velodyne 100" />


    <!--- points_preprocessor_usi -->
    <node pkg="points_preprocessor_usi" type="groundplanfit"    name="groundplanfit"    output="screen" >
        <param name="point_topic" value="/kitti/velo/pointcloud" />
        <!--/velodyne_points-->
        <param name="no_ground_point_topic" value="/points_no_ground" />
        <param name="ground_point_topic" value="/points_ground" />
        <param name="sensor_model" value="64" />
        <param name="sensor_height" value="2.2" />  <!--2.5-->
        <param name="max_iter" value="10" />
        <param name="num_lpr" value="500" />
        <param name="th_seeds" value="0.5" />   <!--1.2-->
        <param name="th_dist" value="0.3" />
        <param name="save_no_ground_cloud" value="false" />
        <param name="output_directory" value="/home/vance/output/kitti07" />
    </node>

    <node pkg="points_preprocessor_usi" type="scanlinerun"      name="scanlinerun"      output="screen">
        <param name="point_topic" value="/all_points" />
        <param name="th_run" value="0.15" />
        <param name="th_merge" value="0.5" />
        <param name="cluster" value="/slr" />
        <param name="sensor_model" value="64" />
    </node>
<!--
    <node pkg="points_preprocessor_usi" type="pc2DM"      name="pc2DM"      output="screen">
        <param name="input_topic" value="/kitti/velo/pointcloud" />
        <param name="output_topic" value="/dense_map" />
        <param name="sensor_model" value="64" />
        <param name="sensor_resolution" value="1800" />
    </node>-->

    <node pkg="points_preprocessor_usi" type="readDataFromFiles"      name="readDataFromFiles"      output="screen">
        <param name="folder_of_bin_files" value="/home/vance/dataset/mm/persion/data" />
    </node>
<!--
    <node pkg="points_preprocessor_usi" type="dynamic_reconfigure_node"      name="dynamic_reconfigure_node"      output="screen"/>
    <node pkg="rqt_reconfigure" type="rqt_reconfigure"      name="rqt_reconfigure"      output="screen"/>-->



</launch>
